### PR TITLE
Core/Packets Opcodes Structure Update

### PR DESCRIPTION
**Pull request by ProjectcoreDevs**
Opcode structure update for :
CMSG_SELL_ITEM,
CMSG_REPAIR_ITEM,
CMSG_ITEM_REFUND_INFO,
CMSG_BUYBACK_ITEM.
And clean tabs ;)

### CODE REVIEW

File: src/server/game/Server/Packets/ItemPackets.cpp [hunk 1]
```diff
@@ -17,6 +17,31 @@
 
 #include "ItemPackets.h"
 
+void WorldPackets::Item::BuyBackItem::Read()
+{
+	_worldPacket >> VendorGUID;
```
Comment (Shauren) [inline src/server/game/Server/Packets/ItemPackets.cpp]: something something tabs something

File: src/server/game/Server/Packets/ItemPackets.cpp [hunk 2]
```diff
@@ -17,6 +17,31 @@
...
+void WorldPackets::Item::ItemRefundInfo::Read()
+{
+	_worldPacket >> ItemGUID;
+}
+
+void WorldPackets::Item::RepairItem::Read()
+{
+	_worldPacket >> NpcGUID;
+	_worldPacket >> ItemGUID;
+	_worldPacket >> UseGuildBank;
```
Comment (joschiwald) [inline src/server/game/Server/Packets/ItemPackets.cpp]: change to: `UseGuildBank = _worldPacket.ReadBit()`

File: src/server/game/Handlers/NPCHandler.cpp [hunk 3]
```diff
@@ -882,18 +877,18 @@ void WorldSession::HandleRepairItemOpcode(WorldPacket& recvData)
...
         if (item)
-            _player->DurabilityRepair(item->GetPos(), true, discountMod, guildBank != 0);
+            _player->DurabilityRepair(item->GetPos(), true, discountMod, packet.UseGuildBank != 0);
     }
     else
     {
-        TC_LOG_DEBUG("network", "ITEM: Repair all items at %s", npcGUID.ToString().c_str());
-        _player->DurabilityRepairAll(true, discountMod, guildBank != 0);
+        TC_LOG_DEBUG("network", "ITEM: Repair all items at %s", packet.NpcGUID.ToString().c_str());
+        _player->DurabilityRepairAll(true, discountMod, packet.UseGuildBank != 0);
```
Comment (joschiwald) [inline src/server/game/Handlers/NPCHandler.cpp]: checking bool != 0 looks bad

**Comment by joschiwald** [comment]
a7956afd6eb278fac26f3aa47df2033176fcfae6
removed tabs

**Pull request closed without merging**